lib/p2p/Pool.ts

```diff
@@ -456,6 +476,14 @@ class Pool extends EventEmitter {
     });
   }
 
+  private clearPendingOutgoingConnection = (nodePubKey?: string) => {
```

Comment:
I don't think we really need this method, just using the `pendingOutgoingConnections.delete` method should be fine. You can use `pendingOutgoingConnections.delete(peer.nodePubKey!)` with the `!` non-null assertion operator in the `'open'` event handler, since we already assert elsewhere that on open we'll have a value for `nodePubKey`.

Reply:
Nice, I was wondering how to do that :+1: